api/src/messaging/receive.py

```diff
@@ -0,0 +1,80 @@
+from channels.generic.websocket import AsyncWebsocketConsumer
+from channels.db import database_sync_to_async
+from django.core.exceptions import ObjectDoesNotExist
+import json
+from .models import Room, Message
+
+
+async def receive_message(self, text_data_json: dict) -> dict:
```

Comment:
Because this in now a helper function and no longer a method, we should be using the self parameter anymore. Any data that this function needs should be passed in as a parameter.